management/vmutils/integration_test.go

```diff
@@ -74,7 +74,22 @@ func TestDeployPlatformCaptureRedeploy(t *testing.T) {
 	}
 
 	t.Logf("Shutting down VM: %s", vmname)
-	vmc.ShutdownRole(vmname, vmname, vmname)
+	if err := Await(client, func() (management.OperationId, error) {
+		return vmc.ShutdownRole(vmname, vmname, vmname)
+	}); err != nil {
+		t.Error(err)
+	}
+
+	for {
+		deployment, err := vmc.GetDeployment(vmname, vmname)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if deployment.RoleInstanceList[0].PowerState == vm.PowerStateStopped {
+			break
+		}
+		time.Sleep(2 * time.Second)
```

Comment:
Done, but for power state, since that's what I need. We can add a method for RoleStatus as well? (in a separate PR?)